Feature user follow

koheitakahashi opened the pull request:
## What I did

- Added function of user follows.
- Added timeline what displayed posts only for people you follow.
- Added mypage what displayed only my posts.
- Internationalized above functions.

Commit:
Added relationships models.

Commit:
Added test data for checking function of following.

Commit:
Added function of following and follower.

Commit:
Added display of books for each user.

Commit:
Added user_id into strongparameter of books.

Commit:
Added timeline and my page.

Commit:
Added links of user page.

Commit:
Fixed a bug that the book edit screen was not displayed.

Commit:
Fixed design of timeline and books lists.

Commit:
Cleaned controllers.

Because the previous controller has other than CRUD. So I split the code of controller and changed the parts related controllers.

Commit:
Shared view files.

Commit:
Internationalized view related follow.

Commit:
Removed unnecessary files and lines.

Commit:
Fixed by rubocop.

Commit:
Fixed label.

Because previous writing is redundant.

Commit:
Fixed how to write if statement.

Because in the previous way of writing, the cell disappeared.

komagata commented on Gemfile:
```diff
@@ -77,3 +77,5 @@ gem "dotenv-rails"
 gem "ruby-debug-ide"
 gem "debase"
+
+gem "faker"
```
これ以外もそうですが、productionなどで必要なければproductionを除外した場所に書くようにしてください〜

koheitakahashi replied:
`group`の記述の意味、全然理解しておりませんでした💦
環境ごとに、グループ化できるんですね。
修正いたしました。ユーザーフォローに関係ないgemはmasterの方で修正して、`rebase master`しました‼️

komagata commented on app/controllers/books_controller.rb:
```diff
@@ -4,7 +4,8 @@ class BooksController < ApplicationController
   def index
-    @books = current_user.books.page params[:page]
+    following_users = current_user.following
+    @books = Book.where(user_id: following_users).or(Book.where(user_id: current_user)).order("created_at DESC").page params[:page]
```
```suggestion
    @books = Book.where(user_id: following_users).or(Book.where(user_id: current_user)).order(created_at: :desc").page params[:page]
```

koheitakahashi replied:
このように渡すこともできるんですね👀
修正いたしました‼️

komagata commented on app/views/shared/_books_list.html.slim:
```diff
@@ -0,0 +1,27 @@
+table.table.table-hover
+  thead.thead-default
+    tr
+      th = t 'activerecord.attributes.book.created_user'
```
human_attribute_nameを使った方がいいかもです。

koheitakahashi replied:
human_attribute_nameを使った方が、すっきり見えますね💦
修正しました。ユーザーフォローと関係ない部分もそのような書き方をしていたので、masterで修正して、`rebase master`しました‼️

komagata commented on app/views/shared/_books_list.html.slim:
```diff
@@ -0,0 +1,27 @@
+        td = book.picture
+        td = link_to "#{t 'link.show'}", book
+        / 改善点：同じif文を2回使っている
+        td = link_to "#{t 'link.edit'}", edit_book_path(book) if current_user.id == book.user_id
```
```suggestion
        td = link_to t('link.edit'), edit_book_path(book) if current_user == book.user
```

koheitakahashi replied:
修正いたしました‼️
他にも同じような書き方をしている部分も修正しました💪

komagata commented on app/views/users/show.html.slim:
```diff
@@ -20,3 +20,18 @@ table.table.table-hover
       td = @user.profile
+    tr
+      th = t('activerecord.attributes.follows.following')
+      td = link_to "#{@user.following.count}", following_index_path
```
```suggestion
      td = link_to @user.following.count, following_index_path
```

koheitakahashi replied:
修正いたしました🙇‍♂️

Commit:
Fixed the parts indicated.

For example, redundant writing of internationalization and where to put the gems were fixed.

komagata commented on app/views/followers/index.html.slim:
```diff
@@ -0,0 +1 @@
+= render "shared/show_follow", title: t('activerecord.attributes.follows.followers')
```
`app/views/application`ディレクトリにファイルを置くとPATHを省略できるため、shared的なものはそこに置くのが良さそうです。

```erb
<%= render 'foo' %>
```

`app/views/application/_foo.html.erb`を読み込む。

koheitakahashi replied:
なるほど。sharedがなんとなく、収まりが悪い気がしていたのですが、そういう風にすれば良かったんですね🤔

ファイルを移動しました💪

Commit:
Fixed the parts indicated.

In particular, the files have been moved so that the path can be omitted.

komagata reviewed (approved):
確認しました、OKですー🙆‍♂️
